bin/cron/deliver_poste_messages

```diff
@@ -245,20 +245,37 @@ class Deliverer
     @templates[name] = Poste::Template.new IO.read(path), engine
   end
 
-  def parse_addresses(string)
-    string.split(', ').map do |user_string|
-      if user_string.include? '<'
-        user_string =~ /^([^<]*)<([^>]*)>$/
-        {name: $1.strip, email: $2}
-      else
-        {name: nil, email: user_string}
-      end
+  def parse_address_string(string)
+    # A comma without surrounding double-quotes isn't necessarily bad, since it
+    # might be like this:
+    #   friendly name, phd <email@code.org>
+    #
+    # Though in such a case, it should really be provided like this:
+    #   "friendly name, phd", <email@code.org>
+    #
+    # But we are alerting just in case a sender has attempted to provide multiple
+    # email addresses like this:
+    #   friendly name <email@code.org>, second person <email2@code.org>
+    #
+    # Because we deprecated support for this pattern, believing that we've never
+    # actually used it.
+    if /,(?=(?:[^\"]*\"[^\"]*\")*[^\"]*$)/ =~ string
+      Honeybadger.notify(
+        error_message: "Comma outside of surrounding double-quotes detected in address string: #{string}"
+      )
+    end
+
+    if string.include? '<'
+      string =~ /^([^<]*)<([^>]*)>$/
```

Comment:
Certainly a less common case than our existing comma issue, but this doesn't handle the possibility of an escaped or quoted `<` in the display name correctly:
```
"I <3 Code" <iheartcode@example.com>
```
I haven't read the whole standard but based on a preliminary scan of [RFC 2822](https://www.ietf.org/rfc/rfc2822.txt) and [this excellent summary thereof](https://cr.yp.to/immhf/addrlist.html) I don't see anything that would prevent angle brackets in a quoted string `display-name` portion of a `name-addr` token.

If we're having trouble finding code to reuse for this purpose, maybe we should open-source a small gem for it.

Reply:
Good call.  Now improved.